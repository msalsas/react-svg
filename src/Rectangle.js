import React, {Component} from 'react';
import PropTypes from 'prop-types';

export default class Rectangle extends Component {
    static propTypes = {
        width: PropTypes.number,
        height: PropTypes.number,
        strokeWidth: PropTypes.number,
        strokeColor: PropTypes.string,
        fillColor: PropTypes.string
    };

    render() {
        let width = this.props.width + this.props.strokeWidth * 2;
        let height = this.props.height + this.props.strokeWidth * 2;
        return (
            <svg width={width} height={height} xmlns="http://www.w3.org/2000/svg">
                <g>
                    <rect x={this.props.strokeWidth} y={this.props.strokeWidth} width={this.props.width} height={this.props.height} className="msalsas-rectangle"
                          fill={this.props.fillColor} strokeWidth={this.props.strokeWidth} stroke={this.props.strokeColor}/>
                </g>
            </svg>
        );
    }
}

Rectangle.defaultProps = {
    width: 150,
    height: 45,
    strokeWidth: 5,
    strokeColor: '#fff',
    fillColor: 'none'
};
