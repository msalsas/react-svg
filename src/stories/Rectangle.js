import React from 'react';
import { storiesOf, action } from '@storybook/react';
import Rectangle from '../Rectangle';

storiesOf('Rectangle', module)
    .add('orange 100x200px, no fill color', () => {
        const props = {
            width: 100,
            height: 200,
            strokeWidth: 5,
            strokeColor: 'orange',
            fillColor: 'none'
        };

        return renderRectangle(props);
    })
    .add('yellow 200x100px, orange fill color', () => {
        const props = {
            width: 200,
            height: 100,
            strokeWidth: 5,
            strokeColor: 'yellow',
            fillColor: 'orange'
        };

        return renderRectangle(props);
    })
    .add('red 100x100px, no fill color', () => {
        const props = {
            width: 100,
            height: 100,
            strokeWidth: 5,
            strokeColor: 'red',
            fillColor: 'none'
        };

        return renderRectangle(props);
    })
    .add('without options', () => {
        const props = {};

        return renderRectangle(props);
    });


function renderRectangle(props) {
    return (
        <div className="svg-wrapper" style={{ background: 'gray' }}>
            <Rectangle {...props} />
        </div>
    );
}