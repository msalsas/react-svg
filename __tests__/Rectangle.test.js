import React from 'react';
import Rectangle from '../src/Rectangle';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('Test Rectangle component', () => {
    const props = {
        width      : 100,
        height     : 200,
        strokeWidth: 5,
        strokeColor: 'orange',
        fillColor  : 'none'
    };
    const rectangle = shallow((<Rectangle {...props}/>));

    it('should be defined', () => {
        expect(rectangle).toBeDefined();
    });
    it('should render correctly', () => {
        expect(rectangle).toMatchSnapshot();
    });
});
