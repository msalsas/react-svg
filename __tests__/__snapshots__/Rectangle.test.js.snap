// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Rectangle component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Rectangle
    fillColor="none"
    height={200}
    strokeColor="orange"
    strokeWidth={5}
    width={100}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <g>
        <rect
          className="msalsas-rectangle"
          fill="none"
          height={200}
          stroke="orange"
          strokeWidth={5}
          width={100}
          x={5}
          y={5}
        />
      </g>,
      "height": 210,
      "width": 110,
      "xmlns": "http://www.w3.org/2000/svg",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <rect
          className="msalsas-rectangle"
          fill="none"
          height={200}
          stroke="orange"
          strokeWidth={5}
          width={100}
          x={5}
          y={5}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "msalsas-rectangle",
          "fill": "none",
          "height": 200,
          "stroke": "orange",
          "strokeWidth": 5,
          "width": 100,
          "x": 5,
          "y": 5,
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      "type": "g",
    },
    "type": "svg",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <g>
          <rect
            className="msalsas-rectangle"
            fill="none"
            height={200}
            stroke="orange"
            strokeWidth={5}
            width={100}
            x={5}
            y={5}
          />
        </g>,
        "height": 210,
        "width": 110,
        "xmlns": "http://www.w3.org/2000/svg",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <rect
            className="msalsas-rectangle"
            fill="none"
            height={200}
            stroke="orange"
            strokeWidth={5}
            width={100}
            x={5}
            y={5}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "msalsas-rectangle",
            "fill": "none",
            "height": 200,
            "stroke": "orange",
            "strokeWidth": 5,
            "width": 100,
            "x": 5,
            "y": 5,
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        "type": "g",
      },
      "type": "svg",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
